### DISCUSSION TITLE
F series and sequential writes

### DISCUSSION THREAD

**Discussion by gmcmicken** [Arnold-n/P1P2MQTT#64] [Q&A]
I think I'm running into an issue where setting two values (for example, operating mode and fan speed) simultaneously results in only one of them working. I'm guessing because of how and when the 38 packet is formed and sent. Is there a way to send multiple values in one request or work in some type of queue?

**Comment by Arnold-n** [comment]
Indeed at the moment there is no queuing implemented, and only one change can be done at a time; subsequent changes will fail (with an error message). But this is a valid request. For F series we could simply allow multiple parallel writes with little effort, but queuing (perhaps on ESP side) seems a better option.

**Comment by Arnold-n** [comment]
Update: I am working on a queuing mechanism where the ESP buffers commands until the ATmega is ready to accept these.  Will become available as part of a next firmware release [but not yet done in v0.9.46].

**Reply by gmcmicken** [reply]
Nice work! I'm now running esphome as both a thermostat and a zone controller that send commands to the P1P2 interface so I've implemented queueing there, but I would imagine other users with direct MQTT climate entities or other setups will appreciate this!

**Comment by Arnold-n** [comment]
Queueing is still not implemented, but v0.9.46 introduces multiple writes in one command, separated by white space, for example `F3800001 38000851`. Also applies for the `E` command in Daikin E-series.

**Comment by Arnold-n** [comment]
And v0.9.53 will support buffering of E and F commands.
